src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+import java.util.List;
+import java.util.Set;
+import java.util.ArrayList;
+
+import org.opensearch.index.analysis.AnalysisRegistry;
+import org.opensearch.action.admin.indices.analyze.AnalyzeAction;
+import org.opensearch.action.admin.indices.analyze.AnalyzeAction.AnalyzeToken;
+import static org.opensearch.action.admin.indices.analyze.TransportAnalyzeAction.analyze;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterParser.parseStringParameter;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterParser.parseDoubleParameter;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterParser.parseIntegerParameter;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterValidator.validateStringParameter;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterValidator.validatePositiveIntegerParameter;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterValidator.validateDoubleParameterWithinRange;
+
+/**
+ * The implementation {@link Chunker} for fixed token length algorithm.
+ */
+public class FixedTokenLengthChunker implements Chunker {
+
+    public static final String ALGORITHM_NAME = "fixed_token_length";
+
+    // field name for each parameter
+    public static final String ANALYSIS_REGISTRY_FIELD = "analysis_registry";
+    public static final String TOKEN_LIMIT_FIELD = "token_limit";
+    public static final String OVERLAP_RATE_FIELD = "overlap_rate";
+    public static final String MAX_TOKEN_COUNT_FIELD = "max_token_count";
+    public static final String TOKENIZER_FIELD = "tokenizer";
+
+    // default values for each parameter
+    private static final int DEFAULT_TOKEN_LIMIT = 384;
+    private static final double DEFAULT_OVERLAP_RATE = 0.0;
+    private static final int DEFAULT_MAX_TOKEN_COUNT = 10000;
+    private static final String DEFAULT_TOKENIZER = "standard";
+
+    // parameter restrictions
+    private static final double OVERLAP_RATE_LOWER_BOUND = 0.0;
+    private static final double OVERLAP_RATE_UPPER_BOUND = 0.5;
+    private static final Set<String> WORD_TOKENIZERS = Set.of(
+        "standard",
+        "letter",
+        "lowercase",
+        "whitespace",
+        "uax_url_email",
+        "classic",
+        "thai"
+    );
```

Comment:
If a user creates their own tokenizer how it will be supported?

Ref for creating a tokenizer: https://www.elastic.co/guide/en/elasticsearch/reference/7.10/analysis-pattern-tokenizer.html#_example_configuration_10

Reply:
These are the word tokenizers in OpenSearch https://opensearch.org/docs/latest/analyzers/tokenizers/index/#word-tokenizers. For text chunking processor, most user take it as a preprocess step before text embedding. The word tokenizers suffices for the user case.

Reply:
> If a user creates their own tokenizer how it will be supported?

We are not supporting user created tokenizer in this PR. The fixed token length algorithm chunks passage according to start and end offset of each token. It is unclear how some edge cases from weird tokenizer would impact this algorithm. We want to be conservative here and only restrict user to word tokenizer in our first release.

Reply:
What is behavior in case of user-created tokenizer? If we ignore such tokenizer silently it's not the best UX. Please make sure this is documented clearly.

Reply:
lets create a github issue around this where propose the feature of user providing custom tokenizer

Reply:
> What is behavior in case of user-created tokenizer? If we ignore such tokenizer silently it's not the best UX. Please make sure this is documented clearly.

Some tokenizer may create overlap token. For user created tokenizers, it is possible that previous token will show up later in the tokenization list, despite that its start and end offset is smaller.